NewForce-Cohort-6/tabloidfullstack-cranberry-sauce
# Lr6 create a post

As an author, I would like to be able to create Posts so I can share my thoughts with the world.

Given a user is in the app
When they select the New Post menu option
Then they should be directed to a page with a form for creating a new post

Given the user has entered the relevant information for a Post
When they click the Save button
Then the Post should be saved to the database
And the creation datetime should be automatically set to the current date and time
And the post should automatically be approved
And the user should be recorded as the author of the Post
And the user should be redirected to the new Post's details page

The "relevant information" for a Post is

Title
Content
Category
Header Image URL (optional)
Publication Date (optional)





Testing Instructions


1. Commit or stash whatever you're currently working on
2. Git fetch --all
3. Git checkout LR6CreateAPost
4. git pull origin LR6CreateAPost
5. [create database here](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/01_Tabloid_Create_DB.sql)
6. [seed data here](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/02_Tabloid_Seed_Data.sql)
7. Start database from inside Visual Studio by pressing the green play button
8. Run npm start from inside Tabloid client folder.
9. Login using an email that you know is there and works for you (example: [foo@bar.com](mailto:foo@bar.com) from seed data)
10. Once logged-in, use the navbar to navigate to post management ("/post")
11. click create post
12. Enter the information in the required fields. Click Add Post
13. Because the post hasn't been published, it won't show in the list. To ensure it has been added to the database, run a SQL query for all post.
14. Check to see if your post has been added to the bottom of the list.
